File: Api/Models/InventoryModel.cs
namespace Api.Models
{
    public class GalaDetails
    {
        public int Id { get; set; }
        public required string GalaNo { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime UpdateDate { get; set; }
    }

    public class QualityDetails
    {
        public int Id { get; set; }
        public required string QualityType { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime UpdateDate { get; set; }
    }


    public class GalaQualitySerialNoTrack
    {
        public int Id { get; set; }
        public int GalaNo { get; set; }
        public int QualityType { get; set; }
        public int SerialNo { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime UpdateDate { get; set; }
    }

    public class GalaQualitySerialNoTrackHistory
    {
        public int Id { get; set; }
        public int GalaNo { get; set; }
        public int QualityType { get; set; }
        public int SerialNo { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime UpdateDate { get; set; }
    }


    public class Inventory
    {
        public int Id { get; set; }
        public int GalaNo { get; set; }
        public int QualityType { get; set; }
        public int SerialNo { get; set; }
        public int LoomNo { get; set; }
        public decimal Meter { get; set; }
        public decimal NetWeight { get; set; }
        public decimal Avg_NetWeight { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime UpdateDate { get; set; }
    }
}

File: Web/src/app/Model/InventoryModel.ts
export interface GalaDetails {
  id: number;
  galaNo: string;
  isActive: boolean;
  createDate: Date;
  updateDate: Date;
}

export interface QualityDetails {
  id: number;
  qualityType: string;
  isActive: boolean;
  createDate: Date;
  updateDate: Date;
}

export interface GalaQualitySerialNoTrack {
  id: number;
  galaNo: number;
  qualityType: number;
  serialNo: number;
  startDate: Date;
  endDate: Date;
  isActive: boolean;
  createDate: Date;
  updateDate: Date;
}

export interface GalaQualitySerialNoTrackHistory {
  id: number;
  galaNo: number;
  qualityType: number;
  serialNo: number;
  startDate: Date;
  endDate: Date;
  isActive: boolean;
  createDate: Date;
  updateDate: Date;
}

export interface Inventory {
  id: number;
  galaNo: number;
  qualityType: number;
  serialNo: number;
  loomNo: number;
  meter: number;
  netWeight: number;
  avg_NetWeight: number;
  isActive: boolean;
  createDate: Date;
  updateDate: Date;
}